packages/components/src/typography/Label/src/Label.tsx

```diff
@@ -15,12 +16,19 @@ import styles from "./Label.module.css";
 export const GlobalLabelCssSelector = "hop-Label";
 
 // TODO: Add necessityIndicator and required Props
-export interface LabelProps extends StyledComponentProps<RACLabelProps> {}
+export interface LabelProps extends StyledComponentProps<RACLabelProps> {
+    /**
+     * Whether the label show a required state.
+     */
+    isRequired?: boolean;
```

Comment:
I see you added isRequired, but I think Alex wanted to add necessityIndicator as well, judging by the comment in this file.
Something like react spectrum:
https://react-spectrum.adobe.com/react-spectrum/TextArea.html#labeling
This way, the user could either have * when required or have the text (Optional) or (Required).
In our case, I'm not sure we'd want to display both, but maybe just (Optional) when like everything is required except one element.

Reply:
I've discussed it with the Design team and they don't envisage any other method than the * for isRequire inputs. I'd seen Alex's comment on the subject and figured we could do it if necessary when he got back.

Reply:
alright, I'm a bit surprised, but ok.